core/src/main/java/com/expedia/www/haystack/client/Tracer.java

```diff
@@ -197,6 +197,11 @@ private SpanContext createDependentContext() {
         }
 
         private SpanContext createContext() {
+            // handle active spans if needed
+            if (references.isEmpty() && !ignoreActive && tracer.activeSpan() != null) {
```

Comment:
I would suggest combining the first two ifs so that there's only one check for references.isEmpty()

Reply:
The dual test is needed.  Calling `asChildOf` doesn't guaranty that `references.isEmpty()` is `false`.